compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -1200,7 +1200,14 @@ impl<'a> Context<'a> {
             !res_typ.contains_slice_element(),
             "ICE: Nested slice result found during ACIR generation"
         );
-        let value = self.array_get_value(&res_typ, block_id, &mut var_index)?;
+        let mut value = self.array_get_value(&res_typ, block_id, &mut var_index)?;
+        if let AcirValue::Var(value_var, typ) = value {
```

Comment:
What would happen if this is `AcirValue::Array`?

Reply:
My thinking was that overflows would only matter for numerical values, which is the case that is handled.

Reply:
Ok I am going to approve for now as to not block it. If I succeed in making a failure for the array case it can be a separate PR